sdk/helper/storagepacker/ids2keys.go

```diff
@@ -0,0 +1,98 @@
+package storagepacker
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"github.com/hashicorp/vault/sdk/helper/cryptoutil"
+	"math"
+	"sort"
+	"strings"
+)
+
+type itemRequest struct {
```

Comment:
Why is this struct needed as opposed to simply using `Item` directly? The only additional info in here is the hash which is can be calculated using the ID.

Reply:
Recalculating the hash every time we needed it seemed like a bad idea. For example, sorting in Key order would become more expensive.

But, yeah, I originally thought it was going to have a pointer to the Bucket too but that turned out to be unnecessary.

Reply:
I changed this to eliminate itemRequest, and just use Item everywhere (which now has a non-exported "key".)